### ISSUE TITLE
jQuery AJAX Call with cache: false for blueprint routes

### ISSUE THREAD

**Issue by richdunajewski** [balderdashy/sails#2461]
Sails v0.10.5, untested in later versions.

When making a jQuery AJAX call with the cache option set to **false**, and using the default blueprint routes, Sails with throw a 500 error.

````
error: Sending 500 ("Server Error") response: 
 Error (E_UNKNOWN) :: Encountered an unexpected error
error: column user._ does not exist
````

jQuery will add **?_=1417624443734** to the URL to force uniqueness so the browser and/or proxies would not cache the response. Sails assumes that the request is for a model attribute with that name, and fails when it does not exist.

While many different methods exist for bypassing caches in this manner (ie ?t=xxx), jQuery's standard method of using an underscore should at least be supported, in my opinion, given the prolific nature of the framework for AJAX requests. Normally this wouldn't be an issue as the developer would have control over Sails to override this behavior, however if the Sails API were public, the client-side developer will encounter nasty server errors when they set the cache flag to false in jQuery and have no way to investigate or resolve the bugs. 

Another solution is to ignore that attribute, and continue returning matching results. This avoids the 500 error, and results will be returned albeit without the predicate being matched.

**Comment by JetFault** [comment]
Bump. Facing this issue with frameworks adding query params. Right now the way I'm fixing it is adding a policy that removes all query params other than 'limit', 'skip', 'sort', 'where', etc. Pretty much rewriting the `req.options.criteria.blacklist` logic. 

I think there should be an option to disable [building the where based on all query params](https://github.com/balderdashy/sails/blob/master/lib/hooks/blueprints/actionUtil.js#L187). If you want to do some where logic, you can rely on sending `?where` as your query param (again this being a config option, so it would be a non-breaking change).

**Comment by JetFault** [comment]
Another pretty bad thing with this is that it sends 500s. 500s are inherently 'I done goofed' for the developer, so you would create extra logging/alerts for 500s. This would allow a user to easily create server errors and flood logs/alerts.

**Comment by JetFault** [comment]
Ping on this. @mikermcneil @sgress454 

A way I thought of not returning 500's for non-existent columns is in `serverError` check what type of error, if it is a certain type of error from DB, just don't send it back (haven't tested this, so I don't know if we get enough info back from waterline on what kind of DB error it is).

And a way to fix the error in general is to only allow a whitelist of `where` clauses you can use (based on your model attributes). This wouldn't let you use columns that aren't in your Model but in DB. Although, one way I think this can be fixed is only whitelist for blueprints and allow on general waterline queries (this would mean it would be a blueprint change and not a waterline change). So, in addition (or instead) with `criteria.blacklist`

**Comment by CWyrtzen** [comment]
Has this been tested in the latest release?

**Comment by sailsbot** [comment]
Thanks for posting, @richdunajewski.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
